src/parsers/manifest/smooth/representationIndex.ts

```diff
@@ -538,5 +547,18 @@ export default class SmoothRepresentationIndex
       for (let i = 0; i < nextSegments.length; i++) {
         _addSegmentInfos(this._index, nextSegments[i], currentSegment);
       }
+
+      // clean segments before time shift buffer depth
+      const { timeShiftBufferDepth } = this._index;
+      const lastPositionEstimate =
+        (performance.now() - this._indexValidityTime) / 1000 +
+        this._initialLastPosition;
+
+      if (timeShiftBufferDepth != null) {
+        const threshold =
+          (lastPositionEstimate - timeShiftBufferDepth) * this._index.timescale;
+        this._index.timeline = this._index.timeline
+          .filter((segment) => (segment.start + segment.duration) >= threshold);
```

Comment:
Instead of filter, it would be more efficient to do a `for` loop and splice the array as they are in starting order